exec/java-exec/src/main/java/org/apache/drill/exec/work/filter/RuntimeFilterManager.java

```diff
@@ -0,0 +1,666 @@
+      RelNode bestNode = ((RelSubset) leftRelNode).getBest();
+      if (bestNode != null) {
+        return findLeftScanPrel(fieldName, bestNode);
+      } else {
+        return null;
+      }
+    } else {
+      List<RelNode> relNodes = leftRelNode.getInputs();
+      RelNode leftNode = relNodes.get(0);
+      return findLeftScanPrel(fieldName, leftNode);
+    }
+  }
+
+  private static boolean containBlockNode(Prel startNode, Prel endNode) {
+    BlockNodeVisitor blockNodeVisitor = new BlockNodeVisitor();
+    startNode.accept(blockNodeVisitor, endNode);
+    return blockNodeVisitor.isEncounteredBlockNode();
+  }
+
+  private void reconstructTextPlan(Pointer<String> textPlan, Map<String, String> mjOpIdPair2runtimeFilter) {
+    if (textPlan != null && textPlan.value != null && !mjOpIdPair2runtimeFilter.isEmpty()) {
+      String[] lines = textPlan.value.split(lineSeparator);
+      Set<String> idPairs = mjOpIdPair2runtimeFilter.keySet();
+      StringBuilder stringBuilder = new StringBuilder();
+      for (String line : lines) {
+        for (String idPair : idPairs) {
+          if (line.startsWith(idPair)) {
+            line = line + " : " + mjOpIdPair2runtimeFilter.get(idPair);
+          }
+        }
+        stringBuilder.append(line).append(lineSeparator);
+      }
+      textPlan.value = stringBuilder.toString();
+    }
+  }
+
+  private void broadcastAggregatedRuntimeFilter(int joinMajorId, UserBitShared.QueryId queryId, List<String> probeFields) {
+    List<CoordinationProtos.DrillbitEndpoint> scanNodeEps = joinMjId2probdeScanEps.get(joinMajorId);
+    int scanNodeMjId = joinMjId2ScanMjId.get(joinMajorId);
+    for (int minorId = 0; minorId < scanNodeEps.size(); minorId++) {
+      BitData.RuntimeFilterBDef.Builder builder = BitData.RuntimeFilterBDef.newBuilder();
+      for (String probeField : probeFields) {
+        builder.addProbeFields(probeField);
+      }
+      BitData.RuntimeFilterBDef runtimeFilterBDef = builder
+        .setQueryId(queryId)
+        .setMajorFragmentId(scanNodeMjId)
+        .setMinorFragmentId(minorId)
+        .build();
+      RuntimeFilterWritable runtimeFilterWritable = new RuntimeFilterWritable();
+      runtimeFilterWritable.setRuntimeFilterBDef(runtimeFilterBDef);
+      runtimeFilterWritable.setData(aggregatedRuntimeFilter.getData());
+
+      CoordinationProtos.DrillbitEndpoint drillbitEndpoint = scanNodeEps.get(minorId);
+
+      DataTunnel dataTunnel = drillbitContext.getDataConnectionsPool().getTunnel(drillbitEndpoint);
+      Consumer<RpcException> exceptionConsumer = new Consumer<RpcException>() {
+        @Override
+        public void accept(final RpcException e) {
+          //logger.error("fail to broadcast a runtime filter to the probe side scan node", e);
+        }
+
+        @Override
+        public void interrupt(final InterruptedException e) {
+          //logger.error("fail to broadcast a runtime filter to the probe side scan node", e);
```

Comment:
should we log this failure here and above ? Looks like if there is any failure in broadcasting aggregated filter we are treating it as minor event since query can still proceed without it.

Reply:
As the runtime filter processing is async , the error log here is not so important. I am not sure to log it or not ,  so I comment  it here.